File: frontend/src/types/Categories.ts
export type Direction = {
    id: number;
    name: string;
};

export type Category = {
    id: number;
    name: string;
    direction_ids: number[];
};

export type Tag = {
    id: number;
    name: string;
    category_id: number;
};

export type CategoriesInfo = {
    directions: Direction[];
    categories: Category[];
    tags: Tag[];
}

File: backend/internal/model/categories.go
package model

type Directions struct {
	Id   uint   `json:"id"`
	Name string `json:"name"`
}

type Categories struct {
	Id           uint   `json:"id"`
	Name         string `json:"name"`
	DirectionIds []uint `json:"direction_ids"`
}

type Tags struct {
	Id         uint   `json:"id"`
	Name       string `json:"name"`
	CategoryId uint   `json:"category_id"`
}

type CategoriesInfo struct {
	Directions []Directions `json:"directions"`
	Categories []Categories `json:"categories"`
	Tags       []Tags       `json:"tags"`
}
